osu.Game/Database/MutableDatabaseBackedStore.cs

```diff
@@ -131,15 +131,15 @@ public void PurgeDeletable(Expression<Func<T, bool>> query = null)
         {
             using (var usage = ContextFactory.GetForWrite())
             {
-                var context = usage.Context;
+                OsuDbContext context = usage.Context;
 
-                var lookup = context.Set<T>().Where(s => s.DeletePending);
+                IQueryable<T> lookup = context.Set<T>().Where(s => s.DeletePending);
 
                 if (query != null) lookup = lookup.Where(query);
 
                 lookup = AddIncludesForDeletion(lookup);
 
-                var purgeable = lookup.ToList();
+                List<T> purgeable = lookup.ToList();
```

Comment:
also here, we're converting the `IQueryable<T>` to list, isn't it obvious that the type would be `List<T>`

Reply:
There are many subclasses of List<T>, so yes, it would not be as obvious as you think.